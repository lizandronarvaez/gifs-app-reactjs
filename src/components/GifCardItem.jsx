import PropTypes from 'prop-types'

const GifCardItem = ({ title, url }) => {

    return (
        <div className='card'>
            <img className='card-image' src={url} alt={title} />
            <p className='card-title'>{title}</p>
        </div>
    )
}

GifCardItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifCardItem