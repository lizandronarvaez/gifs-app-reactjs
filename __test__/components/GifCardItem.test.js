/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react"
import GifCardItem from "../../src/components/GifCardItem";
describe('Pruebas en el componente <GifCardItem />', () => {
    const title = "Personaje";
    const url = "http://localhost/Personaje-url"

    test('Debe realizar un snapshot', () => {

        const { container } = render(<GifCardItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

        // screen.debug()
    });

    test('Debe mostar la imagen con la URL y el nombre de ALT', () => {

        render(<GifCardItem title={title} url={url} />);
        const { src, alt } = screen.getByRole("img");

        expect(src).toBe(url)
        expect(alt).toBe(title)
        // screen.debug()
    });

});