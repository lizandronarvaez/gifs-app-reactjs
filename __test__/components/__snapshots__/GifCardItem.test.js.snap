// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifCardItem /> Debe realizar un snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Personaje"
      class="card-image"
      src="http://localhost/Personaje-url"
    />
    <p
      class="card-title"
    >
      Personaje
    </p>
  </div>
</div>
`;
